docs/09_archival.md

```diff
@@ -0,0 +1,91 @@
+being uploaded to blobstore first. These cases are rare but are possible with the current state of archival.
+
+Work is being prioritized on archival to eliminate these limitations.
+
+## Configuring Archival
+
+Archival is controlled by both domain level config and cluster level config. 
+
+### Cluster Archival Config
+
+A cadence cluster can be in one of three archival states:
+  * **Disabled:** No archivals will occur and the blobstore is not initialized on startup.
+  * **Paused:** This state is not yet implemented. Currently setting cluster to paused is the same as setting it to disabled.
+  * **Enabled:** Archivals will occur.
+
+Enabling the cluster for archival simply means histories are being written to blobstore. There is another
+config which controls if histories can be access from archival. Both these configs have defaults defined in static yaml, and have dynamic config overwrites.
+
+### Domain Archival Config
+
+A domain includes two pieces of archival related config: 
+  * **Status:** Either enabled or disabled. If a domain is in the disabled state no archivals will occur for that domain. 
+  A domain can safely switch between statuses.
+  * **Bucket:** The location in blobstore where histories will be archived into. When a domain enables archival
+  for the first time bucket is set and can never be mutated. If bucket is not specified when first enabling
+  a domain for archival a default bucket name from static config will be used.
+
+## Running Locally
+
+In order to run locally do the following:
+1. `./cadence-server start`
+2. `./cadence --do samples-domain domain register --gd false --archival_status enabled --retention 0`
+3. Run the [helloworld cadence-sample](https://github.com/samarabbas/cadence-samples) by following the README
+4. Copy the workflowID and runID of the completed workflow from log output
+5. `./cadence --do samples-domain wf show --wid <workflowID> --rid <runID>`
+
+This archival setup is using the local filesystem to implement the blobstore. You can see the blobstore is mounted
+out of the `/tmp/development/blobstore` directory. 
+
+Cadence also supports running locally with S3. Follow the README in `common/blobstore/s3store/README.md`.
+
+## FAQ
+
+### How does archival interact with global domains?
+When archival occurs it will first run on the active side and some time later it will run on the standby side as well. 
+Before uploading history a check is done to see if it has already been uploaded, if so it is not re-uploaded.
+
+### What are the supported blobstores?
+All archival code interacts with a blobstore interface. Currently there 
+are two implementations of this interface: S3 and filestore. Filestore is ony used
+for running locally. In order to run on top of a different blobstore all that needs
+to be done is to write a new implementation of the interface backed by the desired blobstore.
+
+### Why are *.index files uploaded along with history files?
```

Comment:
It looks like the * turned on bold for the rest of the doc. Might you need to escape that character here?